crates/transaction-generator-lib/src/call_custom_modules.rs

```diff
@@ -1,46 +1,75 @@
 // Copyright © Aptos Foundation
 // SPDX-License-Identifier: Apache-2.0
 
-use super::{
-    publishing::{module_simple::EntryPoints, publish_util::Package},
-    TransactionExecutor,
-};
+use super::{publishing::publish_util::Package, TransactionExecutor};
 use crate::{
-    publishing::publish_util::PackageHandler, TransactionGenerator, TransactionGeneratorCreator,
+    create_account_transaction, publishing::publish_util::PackageHandler, TransactionGenerator,
+    TransactionGeneratorCreator,
 };
 use aptos_logger::info;
 use aptos_sdk::{
-    move_types::account_address::AccountAddress,
     transaction_builder::TransactionFactory,
     types::{transaction::SignedTransaction, LocalAccount},
 };
+use async_trait::async_trait;
 use rand::{rngs::StdRng, seq::SliceRandom, SeedableRng};
 use std::sync::Arc;
 
-pub struct CallCustomModulesGenerator {
+// Fn + Send + Sync, as it will be called from multiple threads simultaneously
+// if you need any coordination, use Arc<RwLock<X>> fields
+pub type TransactionGeneratorWorker = dyn Fn(
+        &mut LocalAccount,
+        &Package,
+        &LocalAccount,
+        &TransactionFactory,
+        &mut StdRng,
+    ) -> SignedTransaction
+    + Send
+    + Sync;
+
+#[async_trait]
+pub trait UserModuleTransactionGenerator: Sync + Send {
+    fn initialize_package(
```

Comment:
nit: any chance we could add comments to these so it's obvious what they do? `initialize_package` vs `initialize` doesn't tell n00bs like me much. How are they different?